src/mlpack/core/hpt/hpt_impl.hpp

```diff
@@ -0,0 +1,246 @@
+}
+
+template<typename MLAlgorithm,
+         typename Metric,
+         template<typename, typename, typename, typename, typename> class CV,
+         typename Optimizer,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+template<size_t I, typename ArgsTuple, typename... Collections, typename>
+void HyperParameterTuner<MLAlgorithm,
+                         Metric,
+                         CV,
+                         Optimizer,
+                         MatType,
+                         PredictionsType,
+                         WeightsType>::InitGridSearch(
+    const ArgsTuple& /* args */,
+    Collections... collections)
+{
+  optimizer = Optimizer(collections...);
+}
+
+template<typename MLAlgorithm,
+         typename Metric,
+         template<typename, typename, typename, typename, typename> class CV,
+         typename Optimizer,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+template<size_t I, class ArgsTuple, class... Collections, typename, typename>
+void HyperParameterTuner<MLAlgorithm,
+                         Metric,
+                         CV,
+                         Optimizer,
+                         MatType,
+                         PredictionsType,
+                         WeightsType>::InitGridSearch(
+    const ArgsTuple& args,
+    Collections... collections)
+{
+  InitGridSearch<I + 1>(args, collections...);
+}
+
+template<typename MLAlgorithm,
+         typename Metric,
+         template<typename, typename, typename, typename, typename> class CV,
+         typename Optimizer,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+template<size_t I, class ArgsTuple, class... Collections, class, class, class>
+void HyperParameterTuner<MLAlgorithm,
+                         Metric,
+                         CV,
+                         Optimizer,
+                         MatType,
+                         PredictionsType,
+                         WeightsType>::InitGridSearch(
+    const ArgsTuple& args,
+    Collections... collections)
+{
+  InitGridSearch<I + 1>(args, collections..., std::get<I>(args));
+}
+
+template<typename MLAlgorithm,
+         typename Metric,
+         template<typename, typename, typename, typename, typename> class CV,
+         typename Optimizer,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+template<size_t I, typename ArgsTuple, typename... BoundArgs, typename>
+void HyperParameterTuner<MLAlgorithm,
+                         Metric,
+                         CV,
+                         Optimizer,
+                         MatType,
+                         PredictionsType,
+                         WeightsType>::InitCVFunctionAndOptimize(
+    const ArgsTuple& /* args */,
+    arma::mat& bestParams,
+    BoundArgs... boundArgs)
```

Comment:
I would like to make sure - should we prefer the name `Fixed` rather than `Fix` for the function that was called `Bind`?